orchestra/tasks.py

```diff
@@ -0,0 +1,369 @@
+
+
+class GleamRegions(luigi.ExternalTask):
+    gleams = luigi.Parameter(default="data/regions-gleam.csv")
+
+    def output(self):
+        return luigi.LocalTarget(self.gleams)
+
+
+@inherits(RegionsFile, GleamRegions)
+class RegionsDatasetTask(luigi.Task):
+    region_dataset = luigi.Parameter(
+        config_path=default_from_config("RegionsDatasetTask", "region_dataset")
+    )
+
+    def run(self):
+        regions = self.input()["region_file"].path
+        gleams = self.input()["gleam_regions"].path
+        rds = RegionDataset.load(regions, gleams)
+        algorithms.estimate_missing_populations(rds)
+        with open(self.region_dataset, "wb") as ofile:
+            dill.dump(rds, ofile)
+
+    def output(self):
+        return luigi.LocalTarget(self.region_dataset)
+
+    def requires(self):
+        return {
+            "region_file": self.clone(RegionsFile),
+            "gleam_regions": self.clone(GleamRegions),
+        }
+
+    @staticmethod
+    def load_dilled_rds(path: str):
+        with open(path, "rb") as ifile:
+            return dill.load(ifile)
+
+
+@requires(RegionsDatasetTask)
+class JohnsHopkins(luigi.Task):
+    hopkins_output: str = luigi.Parameter(
+        config_path=default_from_config("JohnsHopkins", "hopkins_output")
+    )
+
+    def run(self):
+        log.info("Downloading and parsing CSSE ...")
+        rds = RegionsDatasetTask.load_dilled_rds(self.input().path)
+        csse = imports.import_johns_hopkins(rds)
+        dest = self.hopkins_output
+        csse.to_csv(dest)
+        log.info(
+            f"Saved CSSE to {dest}, last day is {csse.index.get_level_values(1).max()}"
+        )
+
+    def output(self):
+        return luigi.LocalTarget(self.hopkins_output)
+
+
+@requires(RegionsDatasetTask)
+class UpdateForetold(luigi.Task):
+    foretold_output: str = luigi.Parameter(
+        config_path=default_from_config("UpdateForetold", "foretold_output")
+    )
+    foretold_channel: str = luigi.Parameter(
+        config_path=default_from_config("UpdateForetold", "foretold_channel")
+    )
+
+    def run(self):
+        log.info("Downloading and parsing foretold")
+        rds = RegionsDatasetTask.load_dilled_rds(self.input().path)
+        foretold = imports.import_foretold(rds, self.foretold_channel)
+        dest = self.foretold_output
+        foretold.to_csv(dest, float_format="%.7g")
+        log.info(f"Saved Foretold to {dest}")
+
+    def output(self):
+        return luigi.LocalTarget(self.foretold_output)
+
+
+class BaseDefinition(luigi.ExternalTask):
+    base_def: str = luigi.Parameter(default="data/config.xml")
+
+    def output(self):
+        return luigi.LocalTarget(self.base_def)
+
+
+class CountryEstimates(luigi.ExternalTask):
+    country_estimates: str = luigi.Parameter(default="data/country_estimates.csv")
+
+    def output(self):
+        return luigi.LocalTarget(self.country_estimates)
+
+
+class ConfigYaml(luigi.ExternalTask):
```

Comment:
Ideally, this would get replaced by `luigi.cfg` in the future. A lot of implicit parameters has been abstracted to explicit ones for tasks as you can see in the tasks.